src/scripts/validate.py

```diff
@@ -0,0 +1,243 @@
+        counties = sorted(list(stats[state].keys()))
+
+        for county in exp_counties:
+            if county not in counties:
+                # error, not exception, since web app can handle missing counties in statsForMap
+                print("ERROR: statsForMap state {} is missing county {}".format(state, county))
+            else:
+                scenarios = sorted(list(stats[state][county].keys()))
+                scenario_names = scenarios if not scenario_names else scenario_names
+                indicators = set(stats[state][county][scenarios[0]].keys()) if scenarios else None
+
+                if indicators != expected_indicators:
+                    raise Exception("statsForMap {}, {} is missing an expected indicator from {}"
+                        .format(state, county, expected_indicators))
+    
+    return set(scenario_names)
+
+
+def validate_scenario_names(geoid_scenarios: set, map_scenarios: set):
+    ''' Validate scenario names of geoid file match scenario names of statsForMap '''
+    if geoid_scenarios != map_scenarios:
+        raise Exception("County-level geoid scenarios {} do not match statsForMap scenarios {}"
+            .format(geoid_scenarios, map_scenarios))
+
+
+def validate_boundaries(boundaries: dict, expected_states: list, expected_counties: list):
+    ''' Validate countyBoundaries contain expected counties and states '''
+
+    states = sorted(list(boundaries.keys()))
+    for state in expected_states:
+        if state not in states:
+            raise Exception("County boundaries is missing state {}".format(state))
+        county_num = len(boundaries[state]['features'])
+        exp_counties = [x for x in expected_counties if x.startswith(state)]
+
+        for i in range(county_num):
+            county = boundaries[state]['features'][i]['properties']['geoid']
+            if county in exp_counties:
+                exp_counties.remove(county)
+        if exp_counties:
+            print("ERROR: countyBoundaries state {} is missing counties {}"
+                .format(state, exp_counties))
+
+
+def validate_actuals(actuals: list, expected_states: list, expected_counties: list):
+    ''' Validate expected number of county-level and state-level actuals files exists '''
+    counties = [x for x in actuals if len(x) == 5]
+    states = [x for x in actuals if len(x) == 2]
+
+    if len(states) < len(expected_states):
+        raise Exception("Expecting at least {} state-level actual.json files, not {}"
+            .format(len(expected_states), len(states)))
+
+    if len(counties) < len(expected_counties):
+        raise Exception("Expecting at least {} county-level actual.json files, not {}"
+            .format(len(expected_counties), len(counties)))
+
+
+def validate_reported_indicators(actual_obj: dict):
+    ''' 
+    Validate actual file includes expected reported indicators 
+    Confirm date format and value type are '%Y-%m%d' and int, respectively
```

Comment:
should there be a "-" between %m and %d?

Reply:
Yes, thanks